File: features/top_ads/top_ads_dashboard/src/main/java/com/tokopedia/topads/dashboard/view/adapter/TopAdsStatisticPagerAdapter.java
package com.tokopedia.topads.dashboard.view.adapter;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.tokopedia.topads.dashboard.R;

import java.util.List;

/**
 * Created by zulfikarrahman on 1/5/17.
 */

public class TopAdsStatisticPagerAdapter extends FragmentStatePagerAdapter {
    private final List<Fragment> fragmentList;
    private final Context context;

    public TopAdsStatisticPagerAdapter(Context context, FragmentManager fm, List<Fragment> fragmentList) {
        super(fm);
        this.fragmentList = fragmentList;
        this.context = context;
    }

    @Override
    public Fragment getItem(int position) {
        return fragmentList.get(position);
    }

    @Override
    public int getCount() {
        return fragmentList.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return context.getString(R.string.label_top_ads_impression);
            case 1:
                return context.getString(R.string.label_top_ads_click);
            case 2:
                return context.getString(R.string.label_top_ads_ctr);
            case 3:
                return context.getString(R.string.label_top_ads_conversion);
            case 4:
                return context.getString(R.string.label_top_ads_average);
            case 5:
                return context.getString(R.string.label_top_ads_cost);
            default:
                return "";
        }
    }

}

File: features/top_ads/top_ads_dashboard/src/main/java/com/tokopedia/topads/dashboard/view/adapter/TopAdsStatisticPagerAdapter.kt
package com.tokopedia.topads.dashboard.view.adapter

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

import com.tokopedia.topads.dashboard.R

/**
 * Created by zulfikarrahman on 1/5/17.
 */

class TopAdsStatisticPagerAdapter(private val context: Context, fm: FragmentManager, private val fragmentList: List<Fragment>) : FragmentStatePagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return fragmentList[position]
    }

    override fun getCount(): Int {
        return fragmentList.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        when (position) {
            0 -> return context.getString(R.string.label_top_ads_impression)
            1 -> return context.getString(R.string.label_top_ads_click)
            2 -> return context.getString(R.string.label_top_ads_ctr)
            3 -> return context.getString(R.string.label_top_ads_conversion)
            4 -> return context.getString(R.string.label_top_ads_average)
            5 -> return context.getString(R.string.label_top_ads_cost)
            else -> return ""
        }
    }

}
